src/machine/machine_rp2040_rom.go

```diff
@@ -242,3 +243,13 @@ func (f flashBlockDevice) pad(p []byte) []byte {
 	padding := bytes.Repeat([]byte{0xff}, int(paddingNeeded))
 	return append(p, padding...)
 }
+
+// return the correct address to be used for write
+func writeAddress(off int64) uintptr {
+	return FlashDataStart() - uintptr(C.XIP_BASE) + uintptr(off)
```

Comment:
Relies on correct understanding of operator precedence, perhaps re-order as `return FlashDataStart() + uintptr(off) - uintptr(C.XIP_BASE)` or add brackets?

Reply:
Done.